python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -225,23 +315,48 @@ def _match_file(
     return index.query(s_type.hash_from_file(path))
 
 
+def _group_hashes_by_prefix(
+    path: pathlib.Path,
+    settings: CLISettings,
+    hashes_grouped_by_prefix: t.Dict[t.Optional[str], t.Set[str]],
+) -> None:
+    for line in path.read_text().splitlines():
+        line = line.strip()
+        if not line:
+            continue
+        components = line.split()
```

Comment:
I'm torn over what to do when len(component) > 2 - this must mean that there are spaces in the hash, which some future type could allow, but it means our naive parsing here will fail oddly.

Reply:
Completely agree, and had a similar thought -- on re-evaluation I'm thinking for when len(component) > 1 it's probably better to assume that [0] is the prefix and concat [1:] into the hash. 

The only issue here is that if there is a future hash with a space, without a prefix the parsing will fail still. I'll have a think on this

Reply:
I think your intuition is right - if there are hashes with a space, providing the prefix allows for a non-ambiguous parse. Check out `lpartition` or `split(limit=2)`